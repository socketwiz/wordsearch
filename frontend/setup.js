
import Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';

configure({adapter: new Adapter()});

const shallowWithStore = (component, store) => {
    const context = {
        store
    };

    return shallow(component, {context});
};

export default shallowWithStore;
