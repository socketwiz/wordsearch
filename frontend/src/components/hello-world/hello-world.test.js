/* global afterEach describe expect jest it */

import '../../../setup';

import {mount, shallow} from 'enzyme';
import React from 'react';
import HelloWorld from './index';

describe('HelloWorld component', () => {
    it('renders HelloWorld without crashing', () => {
        const wrapper = shallow(<HelloWorld />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders a list without crashing', () => {
        const hellos = [
            {'id': 0, 'message': 'foo'},
            {'id': 1, 'message': 'bar'},
            {'id': 2, 'message': 'baz'}
        ];
        const wrapper = shallow(<HelloWorld hellos={hellos} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should call onSubmit', () => {
        const putMessageMock = (msgObj) => {
            expect(msgObj).toEqual({'message': ''});

            return {
                'catch': (error) => {}
            };
        };
        const component = mount(<HelloWorld putMessage={putMessageMock} />);

        component.find('form').simulate('submit');
    });

    describe('Internal methods', () => {
        let spy;

        afterEach(() => {
            if (spy) {
                spy.mockClear();
            }
        });

        it('should call onSubmit and catch error', () => {
            const putMessageMock = (message) => {
                const promise = new Promise((resolve, reject) => {
                    reject('test-error');
                });
                spy = jest.spyOn(promise, 'catch');

                return promise;
            };
            const component = mount(<HelloWorld putMessage={putMessageMock} />);

            component.find('form').simulate('submit');
            expect(spy).toHaveBeenCalled();
        });

        it('should call getMessage and pass', () => {
            const getMessage = (message) => {
                const promise = new Promise((resolve, reject) => {
                    resolve('test-succeed');
                });
                spy = jest.spyOn(promise, 'catch');

                return promise;
            };
            const component = mount(<HelloWorld getMessage={getMessage} />);

            component.find('form').simulate('submit');
        });

        it('should call getMessage and fail', () => {
            const getMessage = (message) => {
                const promise = new Promise((resolve, reject) => {
                    reject('test-error');
                });
                spy = jest.spyOn(promise, 'catch');

                return promise;
            };
            const component = mount(<HelloWorld getMessage={getMessage} />);

            component.find('form').simulate('submit');
        });

        it('should pass a selected value to the onChange function', () => {
            const component = shallow(<HelloWorld />);

            component.find('input').simulate('change', {
                'currentTarget': {
                    'value': 'some-change'
                }
            });

            expect(component).toMatchSnapshot();
        });
    });
});
