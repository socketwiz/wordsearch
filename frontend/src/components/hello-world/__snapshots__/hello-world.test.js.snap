// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld component Internal methods should pass a selected value to the onChange function 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HelloWorld />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <p>
          <a
                    href="http://localhost:8000/explorer"
          >
                    Rest API Explorer
          </a>
</p>,
        <hr />,
        Array [],
        <hr />,
        <form
          onSubmit={[Function]}
>
          <p>
                    <label
                              htmlFor="exampleInputMessage1"
                    >
                              Message
                    </label>
                    <input
                              aria-describedby="messageHelp"
                              className="form-control"
                              id="exampleInputMessage1"
                              onChange={[Function]}
                              placeholder="Enter message"
                              type="message"
                    />
          </p>
          <button
                    className="btn btn-primary"
                    type="submit"
          >
                    Submit
          </button>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="http://localhost:8000/explorer"
>
            Rest API Explorer
</a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rest API Explorer",
            "href": "http://localhost:8000/explorer",
          },
          "ref": null,
          "rendered": "Rest API Explorer",
          "type": "a",
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <label
                            htmlFor="exampleInputMessage1"
              >
                            Message
              </label>
              <input
                            aria-describedby="messageHelp"
                            className="form-control"
                            id="exampleInputMessage1"
                            onChange={[Function]}
                            placeholder="Enter message"
                            type="message"
              />
</p>,
            <button
              className="btn btn-primary"
              type="submit"
>
              Submit
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="exampleInputMessage1"
>
                  Message
</label>,
                <input
                  aria-describedby="messageHelp"
                  className="form-control"
                  id="exampleInputMessage1"
                  onChange={[Function]}
                  placeholder="Enter message"
                  type="message"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Message",
                  "htmlFor": "exampleInputMessage1",
                },
                "ref": null,
                "rendered": "Message",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-describedby": "messageHelp",
                  "className": "form-control",
                  "id": "exampleInputMessage1",
                  "onChange": [Function],
                  "placeholder": "Enter message",
                  "type": "message",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "btn btn-primary",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <p>
            <a
                        href="http://localhost:8000/explorer"
            >
                        Rest API Explorer
            </a>
</p>,
          <hr />,
          Array [],
          <hr />,
          <form
            onSubmit={[Function]}
>
            <p>
                        <label
                                    htmlFor="exampleInputMessage1"
                        >
                                    Message
                        </label>
                        <input
                                    aria-describedby="messageHelp"
                                    className="form-control"
                                    id="exampleInputMessage1"
                                    onChange={[Function]}
                                    placeholder="Enter message"
                                    type="message"
                        />
            </p>
            <button
                        className="btn btn-primary"
                        type="submit"
            >
                        Submit
            </button>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="http://localhost:8000/explorer"
>
              Rest API Explorer
</a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rest API Explorer",
              "href": "http://localhost:8000/explorer",
            },
            "ref": null,
            "rendered": "Rest API Explorer",
            "type": "a",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                <label
                                htmlFor="exampleInputMessage1"
                >
                                Message
                </label>
                <input
                                aria-describedby="messageHelp"
                                className="form-control"
                                id="exampleInputMessage1"
                                onChange={[Function]}
                                placeholder="Enter message"
                                type="message"
                />
</p>,
              <button
                className="btn btn-primary"
                type="submit"
>
                Submit
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="exampleInputMessage1"
>
                    Message
</label>,
                  <input
                    aria-describedby="messageHelp"
                    className="form-control"
                    id="exampleInputMessage1"
                    onChange={[Function]}
                    placeholder="Enter message"
                    type="message"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Message",
                    "htmlFor": "exampleInputMessage1",
                  },
                  "ref": null,
                  "rendered": "Message",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-describedby": "messageHelp",
                    "className": "form-control",
                    "id": "exampleInputMessage1",
                    "onChange": [Function],
                    "placeholder": "Enter message",
                    "type": "message",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "btn btn-primary",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`HelloWorld component renders HelloWorld without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HelloWorld />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <p>
          <a
                    href="http://localhost:8000/explorer"
          >
                    Rest API Explorer
          </a>
</p>,
        <hr />,
        Array [],
        <hr />,
        <form
          onSubmit={[Function]}
>
          <p>
                    <label
                              htmlFor="exampleInputMessage1"
                    >
                              Message
                    </label>
                    <input
                              aria-describedby="messageHelp"
                              className="form-control"
                              id="exampleInputMessage1"
                              onChange={[Function]}
                              placeholder="Enter message"
                              type="message"
                    />
          </p>
          <button
                    className="btn btn-primary"
                    type="submit"
          >
                    Submit
          </button>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="http://localhost:8000/explorer"
>
            Rest API Explorer
</a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rest API Explorer",
            "href": "http://localhost:8000/explorer",
          },
          "ref": null,
          "rendered": "Rest API Explorer",
          "type": "a",
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <label
                            htmlFor="exampleInputMessage1"
              >
                            Message
              </label>
              <input
                            aria-describedby="messageHelp"
                            className="form-control"
                            id="exampleInputMessage1"
                            onChange={[Function]}
                            placeholder="Enter message"
                            type="message"
              />
</p>,
            <button
              className="btn btn-primary"
              type="submit"
>
              Submit
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="exampleInputMessage1"
>
                  Message
</label>,
                <input
                  aria-describedby="messageHelp"
                  className="form-control"
                  id="exampleInputMessage1"
                  onChange={[Function]}
                  placeholder="Enter message"
                  type="message"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Message",
                  "htmlFor": "exampleInputMessage1",
                },
                "ref": null,
                "rendered": "Message",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-describedby": "messageHelp",
                  "className": "form-control",
                  "id": "exampleInputMessage1",
                  "onChange": [Function],
                  "placeholder": "Enter message",
                  "type": "message",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "btn btn-primary",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <p>
            <a
                        href="http://localhost:8000/explorer"
            >
                        Rest API Explorer
            </a>
</p>,
          <hr />,
          Array [],
          <hr />,
          <form
            onSubmit={[Function]}
>
            <p>
                        <label
                                    htmlFor="exampleInputMessage1"
                        >
                                    Message
                        </label>
                        <input
                                    aria-describedby="messageHelp"
                                    className="form-control"
                                    id="exampleInputMessage1"
                                    onChange={[Function]}
                                    placeholder="Enter message"
                                    type="message"
                        />
            </p>
            <button
                        className="btn btn-primary"
                        type="submit"
            >
                        Submit
            </button>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="http://localhost:8000/explorer"
>
              Rest API Explorer
</a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rest API Explorer",
              "href": "http://localhost:8000/explorer",
            },
            "ref": null,
            "rendered": "Rest API Explorer",
            "type": "a",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                <label
                                htmlFor="exampleInputMessage1"
                >
                                Message
                </label>
                <input
                                aria-describedby="messageHelp"
                                className="form-control"
                                id="exampleInputMessage1"
                                onChange={[Function]}
                                placeholder="Enter message"
                                type="message"
                />
</p>,
              <button
                className="btn btn-primary"
                type="submit"
>
                Submit
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="exampleInputMessage1"
>
                    Message
</label>,
                  <input
                    aria-describedby="messageHelp"
                    className="form-control"
                    id="exampleInputMessage1"
                    onChange={[Function]}
                    placeholder="Enter message"
                    type="message"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Message",
                    "htmlFor": "exampleInputMessage1",
                  },
                  "ref": null,
                  "rendered": "Message",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-describedby": "messageHelp",
                    "className": "form-control",
                    "id": "exampleInputMessage1",
                    "onChange": [Function],
                    "placeholder": "Enter message",
                    "type": "message",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "btn btn-primary",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`HelloWorld component renders a list without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HelloWorld
    hellos={
        Array [
            Object {
              "id": 0,
              "message": "foo",
            },
            Object {
              "id": 1,
              "message": "bar",
            },
            Object {
              "id": 2,
              "message": "baz",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <p>
          <a
                    href="http://localhost:8000/explorer"
          >
                    Rest API Explorer
          </a>
</p>,
        <hr />,
        Array [
          <p>
            foo
</p>,
          <p>
            bar
</p>,
          <p>
            baz
</p>,
        ],
        <hr />,
        <form
          onSubmit={[Function]}
>
          <p>
                    <label
                              htmlFor="exampleInputMessage1"
                    >
                              Message
                    </label>
                    <input
                              aria-describedby="messageHelp"
                              className="form-control"
                              id="exampleInputMessage1"
                              onChange={[Function]}
                              placeholder="Enter message"
                              type="message"
                    />
          </p>
          <button
                    className="btn btn-primary"
                    type="submit"
          >
                    Submit
          </button>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="http://localhost:8000/explorer"
>
            Rest API Explorer
</a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rest API Explorer",
            "href": "http://localhost:8000/explorer",
          },
          "ref": null,
          "rendered": "Rest API Explorer",
          "type": "a",
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "foo",
        },
        "ref": null,
        "rendered": "foo",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "bar",
        },
        "ref": null,
        "rendered": "bar",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "baz",
        },
        "ref": null,
        "rendered": "baz",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <label
                            htmlFor="exampleInputMessage1"
              >
                            Message
              </label>
              <input
                            aria-describedby="messageHelp"
                            className="form-control"
                            id="exampleInputMessage1"
                            onChange={[Function]}
                            placeholder="Enter message"
                            type="message"
              />
</p>,
            <button
              className="btn btn-primary"
              type="submit"
>
              Submit
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="exampleInputMessage1"
>
                  Message
</label>,
                <input
                  aria-describedby="messageHelp"
                  className="form-control"
                  id="exampleInputMessage1"
                  onChange={[Function]}
                  placeholder="Enter message"
                  type="message"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Message",
                  "htmlFor": "exampleInputMessage1",
                },
                "ref": null,
                "rendered": "Message",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-describedby": "messageHelp",
                  "className": "form-control",
                  "id": "exampleInputMessage1",
                  "onChange": [Function],
                  "placeholder": "Enter message",
                  "type": "message",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "btn btn-primary",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <p>
            <a
                        href="http://localhost:8000/explorer"
            >
                        Rest API Explorer
            </a>
</p>,
          <hr />,
          Array [
            <p>
              foo
</p>,
            <p>
              bar
</p>,
            <p>
              baz
</p>,
          ],
          <hr />,
          <form
            onSubmit={[Function]}
>
            <p>
                        <label
                                    htmlFor="exampleInputMessage1"
                        >
                                    Message
                        </label>
                        <input
                                    aria-describedby="messageHelp"
                                    className="form-control"
                                    id="exampleInputMessage1"
                                    onChange={[Function]}
                                    placeholder="Enter message"
                                    type="message"
                        />
            </p>
            <button
                        className="btn btn-primary"
                        type="submit"
            >
                        Submit
            </button>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="http://localhost:8000/explorer"
>
              Rest API Explorer
</a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rest API Explorer",
              "href": "http://localhost:8000/explorer",
            },
            "ref": null,
            "rendered": "Rest API Explorer",
            "type": "a",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "foo",
          },
          "ref": null,
          "rendered": "foo",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "bar",
          },
          "ref": null,
          "rendered": "bar",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "baz",
          },
          "ref": null,
          "rendered": "baz",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                <label
                                htmlFor="exampleInputMessage1"
                >
                                Message
                </label>
                <input
                                aria-describedby="messageHelp"
                                className="form-control"
                                id="exampleInputMessage1"
                                onChange={[Function]}
                                placeholder="Enter message"
                                type="message"
                />
</p>,
              <button
                className="btn btn-primary"
                type="submit"
>
                Submit
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="exampleInputMessage1"
>
                    Message
</label>,
                  <input
                    aria-describedby="messageHelp"
                    className="form-control"
                    id="exampleInputMessage1"
                    onChange={[Function]}
                    placeholder="Enter message"
                    type="message"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Message",
                    "htmlFor": "exampleInputMessage1",
                  },
                  "ref": null,
                  "rendered": "Message",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-describedby": "messageHelp",
                    "className": "form-control",
                    "id": "exampleInputMessage1",
                    "onChange": [Function],
                    "placeholder": "Enter message",
                    "type": "message",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "btn btn-primary",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
